src/util/Constants.ts

```diff
@@ -24,3 +24,33 @@ export const COLOR_PANEL = [
   '#FA7D92',
   '#D598D9',
 ];
+
+/**
+ * We use this attribute to identify which element is the one that the user is touching.
+ * The index is the position of the element in the data array.
+ * This can be either a number (for array-based charts) or a string (for the charts that have a matrix-shaped data).
+ */
+export const DATA_ITEM_INDEX_ATTRIBUTE_NAME = 'data-recharts-item-index';
+/**
+ * We use this attribute to identify which element is the one that the user is touching.
+ * DataKey works here as a kind of identifier for the element. It's not a perfect identifier for ~two~ three reasons:
+ *
+ * 1. There can be two different elements with the same dataKey; we won't know which is it
```

Comment:
whew this is a lot of caveats. I guess _most_ of the time its fine so it'll be better than what it is in 3.x today.. but hmmm. 

But yeah I got no solutions for now...

Reply:
It's better than 2.x too